flask_saml_sso/settings.py

```diff
@@ -0,0 +1,124 @@
+import logging
+
+import flask
+from onelogin.saml2.idp_metadata_parser import OneLogin_Saml2_IdPMetadataParser
+
+logger = logging.getLogger('flask_saml_sso')
+
+
+def _get_saml_sp_settings(app):
+    config = app.config.copy()
+
+    name_id_format = config.setdefault(
+        'SAML_NAME_ID_FORMAT', 'urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified'
+    )
+    force_https = config.setdefault('SAML_FORCE_HTTPS', False)
+
+    cert_file = config.setdefault('SAML_CERT_FILE', None)
+    key_file = config.setdefault('SAML_KEY_FILE', None)
+    requests_signed = config.setdefault('SAML_REQUESTS_SIGNED', False)
+
+    sp_settings = {
+        "sp": {
+            "entityId": flask.url_for(
+                'sso.metadata', _external=True, _scheme='https' if force_https else None
```

Comment:
It is not clear whether this should be `None` or `http`. Consider adding a comment for these three force_https fallbacks